packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts

```
@@ -0,0 +1,355 @@
+        if (connectedContainers.length <= 0) {
+            return;
+        }
+        await createDataStoreForContainer(getRandomContainer());
+    };
+
+    const getTestNode = (id: string): TestNode => {
+        const testNode = testNodes.get(id);
+        assert(testNode !== undefined, `Datastore ${id} should exist in datastores!`);
+        return testNode;
+    };
+
+    const updateReferencedDataStores = (id: string) => {
+        if (!refDataStores.has(id)) {
+            refDataStores.add(id);
+            unrefDataStores.delete(id);
+            for (const childId of getTestNode(id).children) {
+                updateReferencedDataStores(childId);
+            }
+        }
+    };
+
+    const hasReferencedParent = (id: string): boolean => {
+        for (const parentNode of getTestNode(id).parents) {
+            if (refDataStores.has(parentNode)) {
+                return true;
+            }
+        }
+        return false;
+    };
+
+    const updateUnreferencedDataStores = (id: string) => {
+        if (!unrefDataStores.has(id) && !hasReferencedParent(id)) {
+            unrefDataStores.add(id);
+            refDataStores.delete(id);
+            const testNode = getTestNode(id);
+            for (const childId of testNode.children) {
+                updateUnreferencedDataStores(childId);
+            }
+        }
+    };
+
+    const referenceDataStore = (dataStore: TestDataObject, parentDataStore: TestDataObject) => {
+        parentDataStore._root.set(dataStore.id, dataStore.handle);
+        getTestNode(parentDataStore.id).children.add(dataStore.id);
+        getTestNode(dataStore.id).parents.add(parentDataStore.id);
+        if (refDataStores.has(parentDataStore.id)) {
+            updateReferencedDataStores(dataStore.id);
+        }
+    };
+
+    const unreferenceDataStore = (dataStore: TestDataObject, parentDataStore: TestDataObject) => {
+        if (rootDataStores.has(dataStore.id)) {
+            return;
```

Want to throw an error here?